src/T2DMap.cpp

```diff
@@ -610,8 +614,14 @@ void T2DMap::paintEvent(QPaintEvent* e)
     QList<int> oneWayExits;
     TRoom* pPlayerRoom = mpMap->mpRoomDB->getRoom(mpMap->mRoomIdHash.value(mpHost->getName()));
     if (!pPlayerRoom) {
-        p.drawText(_w / 2, _h / 2, "No map or no valid position.");
-
+        p.save();
+        p.setPen(Qt::yellow);
+        p.fillRect(0, 0, width(), height(), Qt::darkRed);
```

Comment:
*sigh* yeah it is a little off-topic - but as I was marking the "No map..." message I remembered being bugged by the layout and colour of the message in the past - and as there are no guarantees about the size (or shape) of the translated message it seemed sensible to use the layout system to centre it properly and allow it to be wrapped if the map was very small rather than specify a _start_ point of half-way across the screen {neglecting the fact the message __has__ a length}...